crates/matrix-sdk-ui/src/timeline/traits.rs

```diff
@@ -43,7 +44,7 @@ pub trait RoomExt {
     /// independent events.
     ///
     /// This is the same as using `room.timeline_builder().build()`.
-    async fn timeline(&self) -> Result<Timeline, timeline::Error>;
+    fn timeline(&self) -> impl Future<Output = Result<Timeline, timeline::Error>> + Send;
```

Comment:
I'm even wondering if we can use `async fn` in traits since we are targeting a recent Rust version. I'm lazy checking though. I'm fine with `-> impl Future`, which is exactly the same.

Reply:
I was coming back here to ensure the `+ Send` bound is present. It is present. Well done :-].

Reply:
Thanks for the review!

The compiler warns if we use `async fn`, and we have some builds with warn-as-errors, so it fails those builds. The reasoning is that using an `async fn` in trait